### PR TITLE
Add subsampling compression

### PR DESCRIPTION
**Pull request by ChengMingZhang-ZTE**
1.Add subsampling compression code
2 Add test cases
3.Add example docs about subsampling compression

Signed-off-by: ChengMingZhang-ZTE <zhang.chengming30@zte.com.cn>

### CODE REVIEW

File: neursafe_fl/python/coordinator/validations.py [hunk 1]
```diff
@@ -140,6 +148,9 @@ def _validate_compression_algorithm(config):
     if config["type"].upper() == SUPPORTED_COMPRESSION_ALGORITHM[0]:
         __validate_quantization_algorithm(config)
 
+    if config["type"].upper() == SUPPORTED_COMPRESSION_ALGORITHM[1]:
+        __validate_subsampling_algorithm(config)
+
```
Comment (tangbo-sh) [inline neursafe_fl/python/coordinator/validations.py:154]: poor readability

File: neursafe_fl/python/libs/compression/const.py [hunk 2]
```diff
@@ -6,4 +6,4 @@
 """
 
 
-SUPPORTED_COMPRESSION_ALGORITHM = ["QUANTIZATION"]
+SUPPORTED_COMPRESSION_ALGORITHM = ["QUANTIZATION", "SUBSAMPLING"]
```
Comment (tangbo-sh) [inline neursafe_fl/python/libs/compression/const.py:9]: use enum or other readability data structure

File: neursafe_fl/python/libs/compression/subsampling.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+
+class SubsamplingCompression(Compression):
+    """Subsampling compression algorithm class definition.
+
+    Compress data by sampling. It will extract subset of data, steps as follows:
+
+    1. Generate a seed and generate a random 0,1 mask matrix, in which the
+       proportion of 0 is the same as the set parameter sampling_rate, because 0
+       means to be reserved, 1 means it will be masked.
```
Comment (tangbo-sh) [inline neursafe_fl/python/libs/compression/subsampling.py:40]: reserved，masked？

**Review by kewenwang-ai — APPROVED** [review]

**Comment by tangbo-sh** [comment]
Add subsampling compression algrithm.

**Pull request merged**